src/Spice86.Core/Emulator/VM/EmulationLoop.cs

```diff
@@ -122,14 +129,34 @@ private void RunOnce() {
         _emulatorBreakpointsManager.CheckExecutionBreakPoints();
         _pauseHandler.WaitIfPaused();
         _cpu.ExecuteNext();
-        _performanceMeasurer.UpdateValue(_cpuState.Cycles);
+        MaybeUpdatePerformanceMeasurements();
         _timer.Tick();
         _dmaController.PerformDmaTransfers();
         _cyclesLimiter.RegulateCycles(_cpuState);
     }
 
+    private void MaybeUpdatePerformanceMeasurements() {
```

Comment:
This should have been done in _performanceMeasurer. Also this should not introduce a new counter but rely on cycles to know when to activate.